src/vs/workbench/parts/files/browser/fileActions.ts

```diff
@@ -555,13 +556,15 @@ export class GlobalNewUntitledFileAction extends Action {
 		id: string,
 		label: string,
 		@IWorkbenchEditorService private editorService: IWorkbenchEditorService,
-		@IUntitledEditorService private untitledEditorService: IUntitledEditorService
+		@IUntitledEditorService private untitledEditorService: IUntitledEditorService,
+		@IConfigurationService private configurationService: IConfigurationService
 	) {
 		super(id, label);
 	}
 
 	public run(): TPromise<any> {
-		const input = this.untitledEditorService.createOrGet();
+		const configuration = this.configurationService.getConfiguration<IFilesConfiguration>();
```

Comment:
I suggest to not add this to the action but actually to the place where untitled editors are being created. There can be other ways how untitled buffers open (e.g. from extensions) and I think a configured default language would also apply there.

I am thinking somewhere inside https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/services/untitled/common/untitledEditorService.ts#L177 to use the configuration if the `modeId` parameter is undefined.